x-pack/plugins/fleet/server/routes/agent/upgrade_handler.ts

```diff
@@ -125,18 +135,30 @@ export const postBulkAgentsUpgradeHandler: RequestHandler<
   }
 };
 
-export const checkVersionIsSame = (version: string, kibanaVersion: string) => {
+export const checkKibanaVersion = (
+  version: string,
+  kibanaVersion: string,
+  shouldBeSame: boolean
```

Comment:
I think we should we remove the `shouldBeSame` parameter and have the same behavior for upgrade and bulk upgrade

Reply:
Sure, I thought that we wanted to keep the behaviour for the upgrade but maybe we should keep them aligned